src/host.js

```diff
@@ -10,9 +10,12 @@ export default class Host extends Port {
 			throw new Error(`Unable to extract origin from "${src}"`);
 		}
 
-		var parent = document.getElementById(id);
-		if(parent === null) {
-			throw new Error(`Could not find parent node with id "${id}"`);
+		var parent = id;
+		if (typeof id === 'string') {
```

Comment:
Sorry @rschick, I wasn't subscribed so didn't notice this until now. I like @dbatiste's suggestion, can you change that first parameter to `parentProvider` which is a function it calls to get the parent? Then the callee can choose whether they want to get the element by ID or if they already have it just return the element. There are likely some tests that will need to be created/updated.